README.md

```diff
@@ -134,6 +137,69 @@ getWithShape({
 }).then( ... )
 ```
 
+### Resolving functions
+
+When the value is a function, its result will be used to resolve the shape.
+`falcor-shapes` accepts an optional second parameter; a resolution function called with `(<resolvable>, <shape>, <key>)`, where;
+
+* `resolvable` is the function specified in the shape
+* `shape` is the shape it is part of
+* `key` the function's key in the shape
+
+```js
+var shapeToSet = require('falcor-shapes')
+
+var sets = shapeToSet(
+  {
+    name: function () {
+      return {
+        first: true,
+        last: true
+      }
+    }
+  },
+  function(resolvable, shape, key) {
+    return resolvable()
+  }
+)
+
+model.get.apply(model, sets).then( ... )
+```
+
+A more complex example:
+
+```js
+var shapeToSet = require('falcor-shapes')
+
+var includeFirstName = true
+
+var sets = shapeToSet(
+  {
+    name: function (withFirstName) {
+      if (withFirstName) {
+        return {
+          first: true,
+          last: true
+        }
+      } else {
+        return {
+          last: true
+        }
+      }
+    }
+  },
+  function(resolvable, shape, key) {
+    if (key === 'name') {
+      return resolvable(includeFirstName)
```

Comment:
The use case is an externally defined shape:

```js
// in some `shape-definition` file
module.export =  {
  name: function (withFirstName) {
    if (withFirstName) {
      return {
        first: true,
        last: true
      }
    } else {
      return {
        last: true
      }
    }
  }
}
```

```
// in some client code
var shapeToSet = require('falcor-shapes')
var theShape = require('./shape-definition')

var includeFirstName = true

shapeToSet(theShape, function(resolvable, shape, key) {
  if (key === 'name') {
    return resolvable(includeFirstName)
  } else {
    return resolvable();
  }
}
```

Another use-case is where the resolver is generated via a factory function (or with `.bind()`, etc).